tests/dataset/test_dataset_write.py

```diff
@@ -0,0 +1,93 @@
+import pathlib
```

Comment:
We need tests for writing into tables that already exist, and reading back from those tables with our database reader methods to see whether the schema was updated properly and works.<

We should also make sure that all the args provided to the write methods are forwarded properly.